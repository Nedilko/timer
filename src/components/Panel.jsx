import PropTypes from 'prop-types'

Panel.propTypes = {
  children: PropTypes.array,
}

function Panel({children}) {
  return (
    <div className='bg-gray-300 rounded-xl shadow-xl p-2'>
      {children}
    </div>
  )
}

export default Panel