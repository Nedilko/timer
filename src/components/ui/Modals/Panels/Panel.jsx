import PropTypes from "prop-types";

Panel.propTypes = {
  children: PropTypes.array,
};

function Panel({ children }) {
  return (
    <div
      role="panel"
      onClick={(e) => e.stopPropagation()}
      className="bg-gray-300 rounded-xl shadow-xl p-2"
    >
      {children}
    </div>
  );
}

export default Panel;
