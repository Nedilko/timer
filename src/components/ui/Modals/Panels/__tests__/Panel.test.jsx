import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Panel from "../Panel";

describe("Panel", () => {
  const e = { stopPropagation: jest.fn() };

  beforeEach(() => {
    e.stopPropagation.mockClear();
  });

  it("should match snapshot", () => {
    render(
      <Panel>
        <div>1</div>
        <div>2</div>
        <div>3</div>
      </Panel>
    );
    expect(screen.getByRole("panel")).toMatchSnapshot();
  });

  it("should stop propogation", () => {
    render(<Panel onClick={e.stopPropagation()}></Panel>);

    userEvent.click(screen.getByRole("panel"));

    expect(e.stopPropagation).toBeCalled();
  });
});
