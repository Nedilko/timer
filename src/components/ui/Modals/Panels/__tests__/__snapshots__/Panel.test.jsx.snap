// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should match snapshot 1`] = `
<div
  class="bg-gray-300 rounded-xl shadow-xl p-2"
  role="panel"
>
  <div>
    1
  </div>
  <div>
    2
  </div>
  <div>
    3
  </div>
</div>
`;
